reports/fennec_dashboard/summarize_csv.ipynb

```diff
@@ -0,0 +1,599 @@
+    "    return \"fennec-v4-\" + operating_mode + suffix + \".csv\"\n",
+    "\n",
+    "def safe_get(p, key, default_value=0):\n",
+    "    \"\"\" Checks if key is in p. If it is, return its value. Otherwise, return default_value. \"\"\"\n",
+    "    return p[key] if key in p else default_value\n",
+    "\n",
+    "def safe_increment(p, key):\n",
+    "    \"\"\" Safely increments p[key]. \"\"\"\n",
+    "    p[key] = safe_get(p, key, 0) + 1\n",
+    "\n",
+    "def is_new_profile(submission_epoch, profile_epoch):\n",
+    "    \"\"\"\n",
+    "    Determines if this is a new profile by checking if the submission date\n",
+    "    equals the profile creation date.\n",
+    "    \"\"\"\n",
+    "    return submission_epoch == profile_epoch\n",
+    "\n",
+    "def get_key(p, segments, report_type):\n",
+    "    \"\"\" Build a key-tuple with the dimensions we want to aggregate. \"\"\"\n",
+    "    dims = []\n",
+    "    \n",
+    "    # Translate the API version to a name.\n",
+    "    if 'osversion' in segments:\n",
+    "        dims.append(android_api_level_to_version(p.get('osversion')))\n",
+    "    else:\n",
+    "        dims.append('all')\n",
+    "\n",
+    "    # Only get some of the countries.\n",
+    "    if 'meta/geoCountry' in segments:\n",
+    "        dims.append(get_country(p.get('meta/geoCountry')))\n",
+    "    else:\n",
+    "        dims.append('all')\n",
+    "\n",
+    "    # Only get some of the countries.\n",
+    "    if 'meta/appUpdateChannel' in segments:\n",
+    "        dims.append(p.get('meta/appUpdateChannel'))\n",
+    "    else:\n",
+    "        dims.append('all')\n",
+    "        \n",
+    "    # Append the date, at last. In the weekly mode, that's the first day of the\n",
+    "    # submission week. In the monthly mode, that's the first day of the month.\n",
+    "    submission_date = dt.datetime.strptime(p.get(\"meta/submissionDate\"), \"%Y%m%d\")\n",
+    "    date_string = \"\"\n",
+    "    if report_type == \"monthly\":\n",
+    "        date_string = submission_date.strftime(\"%Y%m01\")\n",
+    "    else:\n",
+    "        first_day = submission_date - datetime.timedelta(days=((submission_date.weekday() + 1) % 7))\n",
+    "        date_string = first_day.strftime(\"%Y%m%d\")\n",
+    "\n",
+    "    dims.append(date_string)\n",
+    "        \n",
+    "    return tuple(dims)\n",
+    "\n",
+    "def run_query(pings, segments, report_type):\n",
+    "    \"\"\"\n",
+    "    Aggregate the pings over the dimensions in \"segments\". We start by generating a key for each\n",
+    "    ping by chaining the values of the dimensions of interest. If we don't care about a particular\n",
+    "    dimension, its value is set to \"all\".\n",
+    "    All the pings belonging to a key are aggregated together.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # Segment the data by indexing them by dimensions.\n",
+    "    segmented_pings = pings.map(lambda p: (get_key(p, segments, report_type), p))\n",
```

Comment:
It might be a good idea to count these before and after filtering for `profileDate` just to make sure we're not throwing away too much data